src/com/tictacgo/data/Space.java

```diff
@@ -0,0 +1,100 @@
+package com.tictacgo.data;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Space {
+  /**
+   * A List of the Pieces which are in this Space.
+   */
+  private List<Piece> pieces;
+
+  /**
+   * Creates a new Space, initially empty.
+   */
+  public Space() {
+    pieces = new ArrayList<Piece>(2);
+  }
+
+  /**
+   * Adds a Piece into this Space.
+   *
+   * @param piece The Piece to add.
+   */
+  public void addPiece(Piece piece) {
+    pieces.add(piece);
```

Comment:
We need to allow them to be added so the collision detection works properly. (See collision) The logic is:

1. Move all Pieces to new locations, assuming no collisions
2. Resolve all halfway collisions
3. Resolve all full collisions. ---> This involves going through each Space and checking the number of Pieces.